axelrod/strategies/hmm.py

```diff
@@ -0,0 +1,149 @@
+from numpy.random import choice
+
+from axelrod import Actions, Player, init_args, random_choice
+
+C, D = Actions.C, Actions.D
+
+
+def is_stochastic_matrix(m, ep=1e-8):
+    """Checks that the matrix m (a list of lists) is a stochastic matrix."""
+    for i in range(len(m)):
+        for j in range(len(m[i])):
+            if (m[i][j] < 0) or (m[i][j] > 1):
+                return False
+        s = sum(m[i])
+        if abs(1. - s) > ep:
+            return False
+    return True
+
+
+class SimpleHMM(object):
+    """Implementation of a basic Hidden Markov Model. We assume that the
+    transition matrix is conditioned on the opponent's last action, so there
+    are two transition matrices. Emission distributions are stored as Bernoulli
+    probabilities for each state. This is essentially a stochastic FSM.
+
+    https://en.wikipedia.org/wiki/Hidden_Markov_model
+    """
+
+    def __init__(self, transitions_C, transitions_D, emission_probabilities,
+                 initial_state):
+        """
+        Params
+        ------
+        transitions_C and transitions_D are square stochastic matrices:
+            lists of lists with all values in [0, 1] and rows that sum to 1.
+        emission_probabilities is a vector of values in [0, 1]
+        initial_state is an element of range(0, len(emission_probabilities))
+        """
+        self.transitions_C = transitions_C
+        self.transitions_D = transitions_D
+        self.emission_probabilities = emission_probabilities
+        self.state = initial_state
+
+    def is_well_formed(self):
+        if not is_stochastic_matrix(self.transitions_C):
```

Comment:
There are multiple checks here so I left them separate. We can change it if you feel strongly about the form.